docs: update getting-started and httproute guides for v1alpha2

shaneutt opened the pull request:
**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

This updates the examples in the `HTTPRoute` guide for `v1alpha2`, and does some spacing cleanup for the getting started guide (but that guide did not appear to need any special updating for `v1alpha2`).

**Which issue(s) this PR fixes**:

Resolves the first two tasks in https://github.com/kubernetes-sigs/gateway-api/issues/783

robscott commented:
/ok-to-test

robscott commented on site-src/v1alpha2/guides/http-routing.md:
```diff
@@ -33,7 +29,7 @@ Route labels and Gateway label selectors allow Routes and Gateways to be
 bound to each other by their respective owners.
```
The paragraph above this needs to be updated to match v1alpha2 route attachment.

robscott commented on examples/v1alpha2/http-routing/gateway.yaml:
```diff
@@ -0,0 +1,17 @@
+  listeners:
+  - protocol: HTTP
+    port: 80
+    routes:
+      namespaces:
+        from: All
+        selector:
+          matchLabels:
+            gateway: prod-web-gw
+      kinds:
+      - kind: HTTPRoute
```
None of this is necessary

shaneutt replied:
This is now removed with adcf37a71d913be2cb3d14f9d269501346e7ca25, and the follow up is c03ae32d6f1e5dc105e7413f459c4d5dd80c5877

robscott commented on examples/v1alpha2/http-routing/bar-httproute.yaml:
```diff
@@ -0,0 +1,18 @@
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: bar-route
+spec:
```
Need parentRefs

shaneutt replied:
Whoops, yep :+1: d96f473443178e60fc4d15ca732443c25d565ccd

robscott commented on examples/v1alpha2/http-routing/foo-httproute.yaml:
```diff
@@ -0,0 +1,15 @@
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: foo-route
+spec:
```
Need parentRefs

shaneutt replied:
d96f473443178e60fc4d15ca732443c25d565ccd :+1:

robscott reviewed (commented):
Thanks for the work on this @shaneutt!

shaneutt commented:
/assign @shaneutt

robscott reviewed (commented):
Just a few more nits, but otherwise LGTM.

shaneutt commented on site-src/v1alpha2/guides/http-routing.md:
```diff
@@ -26,18 +22,17 @@ Gateway which allows Routes to merge on a Gateway as long as they don't
+must be configured with `ParentRefs` which reference the parent gateway for which
+it should attach to receive traffic. The following example shows how the combination
+of `Gateway` and `HTTPRoute` would be configured to enable a basic level of ingress
+to a [Service][svc] serving HTTP traffic:
```
ah, my bad thanks for the catch, fixed in 7a56a48ea578bbb98d07f7e660ef6ecaab02a3be :+1:

shaneutt commented on site-src/v1alpha2/guides/getting-started.md:
```diff
@@ -1,10 +1,7 @@
 
-!!! danger
-    This page has not been updated for v1alpha2 yet.
-
```
Right, because this is the landing page so this should be the last one to remove its banner, makes sense to me in retrospect.

Given that part of the context here seems to be that it's the pages this page _links to_ that are not updated yet, I added this back in b6a39a62adfacb0269602c965231ecdcd2e2e56c but I changed the wording a bit, lmkwyt :+1:

shaneutt commented on examples/v1alpha2/http-routing/gateway.yaml:
```diff
@@ -0,0 +1,32 @@
+metadata:
+  name: example-gateway-class
+spec:
+  controller: acme.io/gateway-controller
```
dropped in c4d98eefaaa2c493e189dc9f71c784d8696921e3

shaneutt commented on examples/v1alpha2/http-routing/foo-httproute.yaml:
```diff
@@ -0,0 +1,18 @@
+---
```
removed in a01c7d30ffd04c477d5084e3e408c991ef44b6e0

youngnick commented:
LGTM pending CI, nice work @shaneutt!

shaneutt commented:
@robscott I think you had asked earlier for the spec to be removed from an example `GatewayClass` however this ended up failing CI verification for the manifest, so I put that back in. Verification is successfull locally now, and I've rebased against recent `master` changes and squashed. Ready for re-review.

robscott commented:
> I think you had asked earlier for the spec to be removed from an example GatewayClass

@shaneutt sorry for the confusion there! I was actually just suggesting that the GatewayClass resource was unnecessary for an example. Most users won't be creating a new GatewayClass for each Gateway, Route, etc. In the [text before and after the example](https://deploy-preview-786--kubernetes-sigs-gateway-api.netlify.app/v1alpha2/guides/http-routing/), there's no mention of the GatewayClass, so I think it could safely be removed.

shaneutt commented:
> > I think you had asked earlier for the spec to be removed from an example GatewayClass
> 
> @shaneutt sorry for the confusion there! I was actually just suggesting that the GatewayClass resource was unnecessary for that example. Most users won't be creating a new GatewayClass for each Gateway, Route, etc. In the [text before and after the example](https://deploy-preview-786--kubernetes-sigs-gateway-api.netlify.app/v1alpha2/guides/http-routing/), there's no mention of the GatewayClass, so I think it could safely be removed from that example.

Sounds good, that's now removed.

robscott commented:
Thanks!

/lgtm

robscott commented:
/approve
